libnmstate/nm/route.py

```diff
@@ -0,0 +1,150 @@
+#
+# Copyright 2019 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+import logging
+from operator import itemgetter
+from libnmstate.schema import LinuxRoute
+
+from libnmstate import iplib
+
+
+def get_running(client, get_ip_cfg_func, skip_networks):
+    ret = []
+    for active_connection in client.get_active_connections():
```

Comment:
You could have passed the ip_cfg list instead of passing the client and the function pointer.
Not a must, but as only the ipvX modules should know about this function anyway, they could just do the calling there and pass the result down to here:
```
acs_and_ip_confs = [(ac, nmclient.NM.ActiveConnection.get_ip4_config(active_connection)) for ac in client.get_active_connections()]
get_running(acs_and_ip_confs)
```

If you want the actual execution to happen at the iteration point and save some memory, you can build it as a generator.

Reply:
Done.